tests/dialects/test_riscv.py

```diff
@@ -0,0 +1,242 @@
+import pytest
+
+from xdsl.dialects.builtin import IntegerAttr, StringAttr
+from xdsl.dialects.riscv import BEQOp, BGTOp, DirectiveOp, JALOp, LUIOp, LabelOp, MVOp, RegisterAttr, LabelAttr, LBOp, JALROp, SBOp, SCALLOp, SLLOp
+
+
+def test_registerattr():
+    reg1 = RegisterAttr.from_index(13)
+    assert reg1.data.index == 13
+    assert reg1.data.get_abi_name() == "a3"
+
+    reg2 = RegisterAttr.from_name("a3")
+    assert reg2.data.get_abi_name() == "a3"
+    assert reg2.data.index == 13
+
+    reg3 = RegisterAttr.from_register(reg1.data)
+    assert reg3.data.index == 13
+    assert reg3.data.get_abi_name() == "a3"
+
+
+def test_labelattr():
+    str = "amazing_string"
+    label = LabelAttr.from_str(str)
+    assert label.data == str
+
+
+def test_riscv1rd1rs1immoperation():
+    reg0 = RegisterAttr.from_index(0)
+    reg1 = RegisterAttr.from_index(1)
+    val = IntegerAttr.from_int_and_width(42, 64)
+
+    op1 = LBOp.get(reg0, reg1, 42)
+    op2 = LBOp.get(reg0, reg1, val)
+
+    assert op1.rd == reg0
+    assert op1.rs1 == reg1
+    assert op1.immediate == val
+
+    assert op2.rd == reg0
+    assert op2.rs1 == reg1
+    assert op2.immediate == val
+
+
+def test_riscv1rd1rs1offoperation():
+    reg0 = RegisterAttr.from_index(0)
+    reg1 = RegisterAttr.from_index(1)
+    val = IntegerAttr.from_int_and_width(42, 64)
+
+    op1 = JALROp.get(reg0, reg1, 42)
+    op2 = JALROp.get(reg0, reg1, val)
```

Comment:
I am not sure when this should happen, but at some point there needs to be a check that immediates have the correct size (e.g. JALR only has room for a 12 bit signed immediate).

Reply:
That could happen at xDSL Op build time, or at assembly output time. I think the latter makes more sense, so that users can be flexible with that and enforce that property with passes at some point. Any thoughts on that?